crates/iroha_wasm_builder/src/lib.rs

```diff
@@ -181,17 +214,27 @@ mod internal {
 
         pub fn build(self) -> Result<Output> {
             let absolute_path = self.absolute_path.clone();
-            self.build_smartcontract().wrap_err_with(|| {
+            let optimize = Profile::is_optimized(self.profile);
+            let output = self.build_smartcontract().wrap_err_with(|| {
                 format!(
                     "Failed to build the smartcontract at path: {}",
                     absolute_path.display()
                 )
-            })
+            })?;
+
+            if optimize {
+                output.optimize()
+            } else {
+                Ok(output)
+            }
```

Comment:
Doesn't this duplicate the optimization in the outer build? Also, this would be inconsistent with the fact that this inner build is used in `build_unoptimized`